src/python/pants/backend/python/lint/bandit/rules.py

```diff
@@ -110,21 +92,11 @@ async def bandit_lint_partition(
             argv=generate_argv(source_files, bandit),
             input_digest=input_digest,
             description=f"Run Bandit on {pluralize(len(partition.field_sets), 'file')}.",
-            output_directories=(LINTER_REPORT_DIR,),
+            output_directories=(REPORT_DIR,),
             level=LogLevel.DEBUG,
         ),
     )
-
-    report: LintReport | None = None
-    report_snapshot = await Get(
-        Snapshot, DigestSubset(result.output_digest, PathGlobs([f"{LINTER_REPORT_DIR}/**"]))
-    )
-    if report_snapshot.files:
-        report_snapshot = await Get(
-            Snapshot, RemovePrefix(report_snapshot.digest, LINTER_REPORT_DIR)
-        )
-        report = LintReport(report_snapshot.digest)
-
+    report = await Get(Digest, RemovePrefix(result.output_digest, REPORT_DIR))
```

Comment:
I believe `DigestSubset` was being used to defensively program around the case of adding more `output_{directories,files}` in the future. But this made things harder to follow and resulted in lots of duplication across the 3 linters/typecheckers. Instead, I think it's better to keep it simple.